### PR TITLE
Add convenience constructor for Close frames

### LINKED ISSUE #8: Add convience constructor for Close frame
Per discussion in the http4s gitter channel, it would be great to have an easier way to set the close code when ending the websocket connection from the server as presently if the code is to be stored as a human readable decimal (say `1000`) it must then be manually converted to an unsigned short in byte array form for use with the existing `Close(data: Array[Byte])` constructor.  

I was about to start working on a PR for this however I'm not sure what form an alternate constructor should take.  Specifically, should the returned `Close` be wrapped in an `Either` so that in the case that the parameter is not one of the pre-defined control codes (which seem to range from 1000 to 4999) we can pass back an error to the caller? Also given that by my reading of the spec the code _should_ never be greater than 4999 even though it is represented as a 16 bit unsigned int, would `short` be an appropriate parameter type for this code, along with an optional `String` for the reason?

### PR DESCRIPTION
**Pull request by keirlawson**
Fixes #8 

Had to bump the SBT version slightly to get it to play nice with the SBT dependecy graph plugin I have installed globally for 0.13, but this can be reversed if required.

### CODE REVIEW

File: project/build.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
-sbt.version=0.13.9
+sbt.version=0.13.10
```
Comment (rossabaker) [inline project/build.properties:1]: Isn't there an 0.13.16?

File: shared/src/test/scala/org/http4s/websocket/WebsocketSpec.scala [hunk 2]
```diff
@@ -72,6 +72,24 @@ class WebsocketSpec extends Specification {
       new String(msg.data, UTF_8) should_== "Hello"
     }
 
+    "encode a close message" in {
+      val frame = Close(1000, "some reason").right.get
+      val msg = decode(encode(frame, true), false)
```
Comment (rossabaker) [inline shared/src/test/scala/org/http4s/websocket/WebsocketSpec.scala]: This would be a great place for a property test.

**Review by rossabaker — APPROVED** [review]

File: project/build.properties (hunk 1 above)
Comment (keirlawson) [reply project/build.properties:1]: There's 0.13.17 infact, but bumping to it introduced a bunch of deprecation warnings and I didn't want to do lots of SBT refactoring so went more conservative

File: shared/src/test/scala/org/http4s/websocket/WebsocketSpec.scala (hunk 2 above)
Comment (keirlawson) [reply shared/src/test/scala/org/http4s/websocket/WebsocketSpec.scala]: Sure, guess I'd better learn how to do that then :D

File: project/build.properties (hunk 1 above)
Comment (rossabaker) [reply project/build.properties:1]: That's fair.  This project might get folded into http4s, if standalone blaze isn't going to do websockets anymore (paging @bryce-anderson), so let's not fight that.

File: shared/src/test/scala/org/http4s/websocket/WebsocketSpec.scala (hunk 2 above)
Comment (rossabaker) [reply shared/src/test/scala/org/http4s/websocket/WebsocketSpec.scala]: I think we can merge without that, but if you try this and get stuck, I'm happy to help.

**Comment by keirlawson** [comment]
Have added scalacheck tests, which prompted me to realise I also need to validate that the reason is no more than 123 bytes of UTF-8. I struggled to create a generator for String between 0 and 123 bytes that produced valid UTF-8 every time so settled for producing strings of 0 to 30 chars, which will produce string of up to 120 bytes, though the distribution will not be even.

File: shared/src/main/scala/org/http4s/websocket/WebsocketBits.scala [hunk 3]
```diff
@@ -138,5 +138,37 @@ object WebsocketBits {
     }
   }
 
+  sealed trait InvalidCloseDataException extends RuntimeException
+  class InvalidCloseCodeException(i: Int) extends InvalidCloseDataException
+  class ReasonTooLongException(s: String) extends InvalidCloseDataException
+
+  private def toUnsigned(x: Int) = Array[Byte](((x >> 8)& 0xFF).toByte, (x & 0xFF).toByte)
```
Comment (bryce-anderson) [inline shared/src/main/scala/org/http4s/websocket/WebsocketBits.scala]: ffti: I would call this `toUnsignedShort`

Should there be   a space between `8)& >>`?

**Review by bryce-anderson — APPROVED** [review]
lgtm.

I haven't spent a lot of time reading the spec lately so I didn't check those specifics.
Regarding pulling into http4s proper, that also seems reasonable to if you think it solves a problem. That said, while I don't _plan_ on supporting websockets in raw blaze, we shouldn't make it more difficult should we change our minds.

**Pull request merged**